src/algorithm/regressor.hxx

```diff
@@ -7,51 +7,209 @@
+  {
+    typedef ModelTpl<Scalar,Options,JointCollectionTpl> Model;
+    typedef DataTpl<Scalar,Options,JointCollectionTpl> Data;
+    
+    typedef boost::fusion::vector<const Model &,
+                                  Data &,
+                                  const JointIndex,
+                                  Eigen::Matrix<Scalar,6,10,Options> &
+                                  > ArgsType;
+    
+    template<typename JointModel>
+    static void algo(const JointModelBase<JointModel> & jmodel,
+                     JointDataBase<typename JointModel::JointDataDerived> & jdata,
+                     const Model & model,
+                     Data & data,
+                     const JointIndex col_idx,
+                     Eigen::Matrix<Scalar,6,10,Options> & A)
+    {
+      typedef typename Model::JointIndex JointIndex;
       
-      forwardKinematics(model,data,q.derived());
+      const JointIndex i = jmodel.id();
+      const JointIndex parent = model.parents[i];
       
-      // Computes the total mass of the system
-      Scalar mass = Scalar(0);
-      for(JointIndex i = 1; i < (JointIndex)model.njoints; ++i)
-        mass += model.inertias[(JointIndex)i].mass();
+      data.jointTorqueRegressor.block(jmodel.idx_v(),10*(int(col_idx)-1),jmodel.nv(),10) = jdata.S().transpose()*A;
       
-      const Scalar mass_inv = Scalar(1)/mass;
-      for(JointIndex i = 1; i < (JointIndex)model.njoints; ++i)
+      if(parent>0)
+          forceSet::se3Action(data.liMi[i],A,A);
+    }
+  };
+
+  template<typename Scalar, int Options, template<typename,int> class JointCollectionTpl, typename ConfigVectorType, typename TangentVectorType1, typename TangentVectorType2>
+  inline typename DataTpl<Scalar,Options,JointCollectionTpl>::MatrixXs &
+  computeJointTorqueRegressor(const ModelTpl<Scalar,Options,JointCollectionTpl> & model,
+                              DataTpl<Scalar,Options,JointCollectionTpl> & data,
+                              const Eigen::MatrixBase<ConfigVectorType> & q,
+                              const Eigen::MatrixBase<TangentVectorType1> & v,
+                              const Eigen::MatrixBase<TangentVectorType2> & a)
+  {
+    assert(model.check(data) && "data is not consistent with model.");
+    assert(q.size() == model.nq);
+    assert(v.size() == model.nv);
+    assert(a.size() == model.nv);
+
+    data.v[0].setZero();
+    data.a_gf[0] = -model.gravity;
+    data.jointTorqueRegressor.setZero();
+
+    typedef RneaForwardStep<Scalar,Options,JointCollectionTpl,ConfigVectorType,TangentVectorType1,TangentVectorType2> Pass1;
+    typename Pass1::ArgsType arg1(model,data,q.derived(),v.derived(),a.derived());
+    for(JointIndex i=1; i<(JointIndex)model.njoints; ++i)
+    {
+      Pass1::run(model.joints[i],data.joints[i],
+                 arg1);
+    }
+
+    typedef JointTorqueRegressorBackwardStep<Scalar,Options,JointCollectionTpl> Pass2;
+    Eigen::Matrix<Scalar,6,10,Options> jointRegressor;
```

Comment:
Avoid memory allocation. Store this matrix in data.

Reply:
My reasoning was that, since this is a fixed-size matrix, it was ok to construct it inside the method itself. Plus I am always reusing the same matrix, within the method. Do you think it is important? I thought if I could avoid putting an extra temporary instance variable inside data, it would be better

Reply:
It will do stack allocation here I guess, because of the size which is larger than a certain value and Eigen will force it.

Reply:
Oh. Ok. But then I should probably rewrite the methods `bodyRegressor`, `jointBodyRegressor` and `frameBodyRegressor` so that they have versions not doing allocation either, I think, otherwise we will have the same problem. `bodyRegressor` and `jointBodyRegressor`, in particular, are employed by `jointTorqueRegressor`.

Is it the still worth it to employ `Eigen::Matrix<Scalar,6,10,Options>`, rather than `Eigen::Matrix<Scalar,Dynamic,Dynamic,Options>`?

Reply:
> Is it the still worth it to employ `Eigen::Matrix<Scalar,6,10,Options>`, rather than `Eigen::Matrix<Scalar,Dynamic,Dynamic,Options>`?

Yes, for static dimensions checking and unrolling of operations.

Reply:
> Oh. Ok. But then I should probably rewrite the methods `bodyRegressor`, `jointBodyRegressor` and `frameBodyRegressor` so that they have versions not doing allocation either, I think, otherwise we will have the same problem. `bodyRegressor` and `jointBodyRegressor`, in particular, are employed by `jointTorqueRegressor`.

I do agree. Thanks a lot.

Reply:
Done. I added a typedef `Data::BodyRegressorType` and a field `data.bodyRegressor`, which is employed by `jointBodyRegressor` and `frameBodyRegressor` to store their results, and by `computeJointTorqueRegressor` as an auxiliary variable.
Let me know if you like the solution.
Wait before you merge, as I still have to fix my NOTE 1.

Reply:
I think a matrix with 60 doubles is far below the stack allocation limit. From Eigen source code:
```cpp
#ifndef EIGEN_STACK_ALLOCATION_LIMIT
// 131072 == 128 KB
#define EIGEN_STACK_ALLOCATION_LIMIT 131072
#endif
```

Reply:
Well, if it is not necessary, I would rather revert the changes

Reply:
As you want. But because of Variant, it will allocate the memory 60 doubles x number of joints types.

Reply:
What about this? Should I revert the changes? In which case, I will ask you to leave the PR open

Reply:
> As you want. But because of Variant, it will allocate the memory 60 doubles x number of joints types.

Mmmh, I understand... Then, we may keep the changes